radix-engine/src/kernel/kernel.rs

```diff
@@ -307,7 +287,7 @@ where
                     }
                 }
                 api.kernel_drop_lock(handle)?;
-                Ok(())
+                api.current_frame.remove_node(&mut api.heap, node_id)
```

Comment:
This part isn't recursive. This is the actual dropping of the node from the call frame (this part is unchanged from before I just moved this statement into the match statement rather than being outside of it).

Reply:
Ha, I misplaced the comment. I was referring to auth zone stack drop which recursively calls proof drop logic, and worktop drop which recursively calls bucket drop logic. 

Like said in the other comment, I think we just need to refactor the drop mechanism a bit.